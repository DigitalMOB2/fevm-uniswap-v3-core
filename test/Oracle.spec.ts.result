

  Oracle
    #initialize
      ✔ index is 0
      ✔ cardinality is 1
      ✔ cardinality next is 1
      ✔ sets first slot timestamp only
      - gas
    #grow
      ✔ increases the cardinality next for the first call
      ✔ does not touch the first slot
      ✔ is no op if oracle is already gte that size
      ✔ adds data to all the slots
      ✔ grow after wrap
      - gas for growing by 1 slot when index == cardinality - 1
      - gas for growing by 10 slots when index == cardinality - 1
      - gas for growing by 1 slot when index != cardinality - 1
      - gas for growing by 10 slots when index != cardinality - 1
    #write
      ✔ single element array gets overwritten
      ✔ does nothing if time has not changed
      ✔ writes an index if time has changed
      ✔ grows cardinality when writing past
      ✔ wraps around
      ✔ accumulates liquidity
    #observe
      before initialization
        ✔ fails before initialize
        1) fails if an older observation does not exist
        ✔ does not fail across overflow boundary
        ✔ interpolates correctly at max liquidity
        ✔ interpolates correctly at min liquidity
        ✔ interpolates the same as 0 liquidity for 1 liquidity
        ✔ interpolates correctly across uint32 seconds boundaries
        ✔ single observation at current time
        2) single observation in past but not earlier than secondsAgo
        ✔ single observation in past at exactly seconds ago
        ✔ single observation in past counterfactual in past
        ✔ single observation in past counterfactual now
        ✔ two observations in chronological order 0 seconds ago exact
        ✔ two observations in chronological order 0 seconds ago counterfactual
        ✔ two observations in chronological order seconds ago is exactly on first observation
        ✔ two observations in chronological order seconds ago is between first and second
        ✔ two observations in reverse order 0 seconds ago exact
        ✔ two observations in reverse order 0 seconds ago counterfactual
        ✔ two observations in reverse order seconds ago is exactly on first observation
        ✔ two observations in reverse order seconds ago is between first and second
        ✔ can fetch multiple observations
        ✔ gas for observe since most recent
        ✔ gas for single observation at current time
        ✔ gas for single observation at current time counterfactually computed
      initialized with 5 observations with starting time of 5
        ✔ index, cardinality, cardinality next
        ✔ latest observation same time as latest
        ✔ latest observation 5 seconds after latest
        ✔ current observation 5 seconds after latest
        ✔ between latest observation and just before latest observation at same time as latest
        ✔ between latest observation and just before latest observation after the latest observation
        3) older than oldest reverts
        ✔ oldest observation
        ✔ oldest observation after some time
        ✔ fetch many values
        ✔ gas all of last 20 seconds
        ✔ gas latest equal
        ✔ gas latest transform
        ✔ gas oldest
        ✔ gas between oldest and oldest + 1
        ✔ gas middle
      initialized with 5 observations with starting time of 4294967291
        ✔ index, cardinality, cardinality next
        ✔ latest observation same time as latest
        ✔ latest observation 5 seconds after latest
        ✔ current observation 5 seconds after latest
        ✔ between latest observation and just before latest observation at same time as latest
        ✔ between latest observation and just before latest observation after the latest observation
        4) older than oldest reverts
        ✔ oldest observation
        ✔ oldest observation after some time
        ✔ fetch many values
        ✔ gas all of last 20 seconds
        ✔ gas latest equal
        ✔ gas latest transform
        ✔ gas oldest
        ✔ gas between oldest and oldest + 1
        ✔ gas middle
    full oracle
      - has max cardinality next
      - has max cardinality
      - index wrapped around
      - can observe into the ordered portion with exact seconds ago
      - can observe into the ordered portion with unexact seconds ago
      - can observe at exactly the latest observation
      - can observe at exactly the latest observation after some time passes
      - can observe after the latest observation counterfactual
      - can observe into the unordered portion of array at exact seconds ago of observation
      - can observe into the unordered portion of array at seconds ago between observations
      - can observe the oldest observation 13*65534 seconds ago
      - can observe the oldest observation 13*65534 + 5 seconds ago if time has elapsed
      - gas cost of observe(0)
      - gas cost of observe(200 * 13)
      - gas cost of observe(200 * 13 + 5)
      - gas cost of observe(0) after 5 seconds
      - gas cost of observe(5) after 5 seconds
      - gas cost of observe(oldest)
      - gas cost of observe(oldest) after 5 seconds


Snapshot Summary
 › 4 snapshots written from 1 test suite.
 › 5 snapshots obsolete from 1 test suite. To remove them all, re-run mocha with `--update` to update them.
   ↳ test/Oracle.spec.ts
       • Oracle #grow gas for growing by 1 slot when index != cardinality - 1 1
       • Oracle #grow gas for growing by 1 slot when index == cardinality - 1 1
       • Oracle #grow gas for growing by 10 slots when index != cardinality - 1 1
       • Oracle #grow gas for growing by 10 slots when index == cardinality - 1 1
       • Oracle #initialize gas 1

  67 passing
  24 pending
  4 failing

  1) Oracle
       #observe
         before initialization
           fails if an older observation does not exist:
     AssertionError: Expected transaction to be reverted with OLD, but other exception was thrown: Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"from":"0xF113C0c7741E1351A6263c7e8cF693494BCf58F7","to":"0x6d77C4530A6Ad90B6301e1eE731fBbf9F54D1c9A","data":"0x883bdbfd000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000001","accessList":null}, error={"name":"ProviderError","_stack":"ProviderError: HttpProviderError\n    at HttpProvider.request (/Users/alex/Documents/Work/mob/filecoin/uniswap-v3-core/node_modules/hardhat/src/internal/core/providers/http.ts:83:19)\n    at LocalAccountsProvider.request (/Users/alex/Documents/Work/mob/filecoin/uniswap-v3-core/node_modules/hardhat/src/internal/core/providers/accounts.ts:187:34)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async EthersProviderWrapper.send (/Users/alex/Documents/Work/mob/filecoin/uniswap-v3-core/node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)","code":1,"_isProviderError":true}, code=CALL_EXCEPTION, version=providers/5.7.2)
  

  2) Oracle
       #observe
         before initialization
           single observation in past but not earlier than secondsAgo:
     AssertionError: Expected transaction to be reverted with OLD, but other exception was thrown: Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"from":"0xF113C0c7741E1351A6263c7e8cF693494BCf58F7","to":"0x457344e8Fb4e7a4688C6Ba2216eB9cC99d2092A1","data":"0x883bdbfd000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000004","accessList":null}, error={"name":"ProviderError","_stack":"ProviderError: HttpProviderError\n    at HttpProvider.request (/Users/alex/Documents/Work/mob/filecoin/uniswap-v3-core/node_modules/hardhat/src/internal/core/providers/http.ts:83:19)\n    at LocalAccountsProvider.request (/Users/alex/Documents/Work/mob/filecoin/uniswap-v3-core/node_modules/hardhat/src/internal/core/providers/accounts.ts:187:34)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async EthersProviderWrapper.send (/Users/alex/Documents/Work/mob/filecoin/uniswap-v3-core/node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)","code":1,"_isProviderError":true}, code=CALL_EXCEPTION, version=providers/5.7.2)
  

  3) Oracle
       #observe
         initialized with 5 observations with starting time of 5
           older than oldest reverts:
     AssertionError: Expected transaction to be reverted with OLD, but other exception was thrown: Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"from":"0xF113C0c7741E1351A6263c7e8cF693494BCf58F7","to":"0x13887C1Cc251FFE42d4E4c7B68C910Cc953b2403","data":"0x883bdbfd00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000f","accessList":null}, error={"name":"ProviderError","_stack":"ProviderError: HttpProviderError\n    at HttpProvider.request (/Users/alex/Documents/Work/mob/filecoin/uniswap-v3-core/node_modules/hardhat/src/internal/core/providers/http.ts:83:19)\n    at LocalAccountsProvider.request (/Users/alex/Documents/Work/mob/filecoin/uniswap-v3-core/node_modules/hardhat/src/internal/core/providers/accounts.ts:187:34)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async EthersProviderWrapper.send (/Users/alex/Documents/Work/mob/filecoin/uniswap-v3-core/node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)","code":1,"_isProviderError":true}, code=CALL_EXCEPTION, version=providers/5.7.2)
  

  4) Oracle
       #observe
         initialized with 5 observations with starting time of 4294967291
           older than oldest reverts:
     AssertionError: Expected transaction to be reverted with OLD, but other exception was thrown: Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"from":"0xF113C0c7741E1351A6263c7e8cF693494BCf58F7","to":"0xb9C4776Db18BBc7B2b612D0FC1c26fEEf8dE47dB","data":"0x883bdbfd00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000f","accessList":null}, error={"name":"ProviderError","_stack":"ProviderError: HttpProviderError\n    at HttpProvider.request (/Users/alex/Documents/Work/mob/filecoin/uniswap-v3-core/node_modules/hardhat/src/internal/core/providers/http.ts:83:19)\n    at LocalAccountsProvider.request (/Users/alex/Documents/Work/mob/filecoin/uniswap-v3-core/node_modules/hardhat/src/internal/core/providers/accounts.ts:187:34)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async EthersProviderWrapper.send (/Users/alex/Documents/Work/mob/filecoin/uniswap-v3-core/node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)","code":1,"_isProviderError":true}, code=CALL_EXCEPTION, version=providers/5.7.2)
  



