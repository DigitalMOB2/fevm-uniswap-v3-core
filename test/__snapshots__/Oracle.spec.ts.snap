// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Oracle #grow gas for growing by 1 slot when index != cardinality - 1 1`] = `7773774`;

exports[`Oracle #grow gas for growing by 1 slot when index == cardinality - 1 1`] = `7482122`;

exports[`Oracle #grow gas for growing by 10 slots when index != cardinality - 1 1`] = `13541937`;

exports[`Oracle #grow gas for growing by 10 slots when index == cardinality - 1 1`] = `13243745`;

exports[`Oracle #initialize gas 1`] = `6600666`;
